addons/stock/models/stock_picking.py

```diff
@@ -329,6 +329,9 @@ class Picking(models.Model):
         'res.partner', 'Contact',
         check_company=True,
         states={'done': [('readonly', True)], 'cancel': [('readonly', True)]})
+    partner_company_id = fields.Many2one(
+        "res.partner", related="partner_id.parent_id", store=True, readonly=True
```

Comment:
Changes on stable versions must respect some rules
The most popular one: we can not add stored field
You will find the entire list here: https://github.com/odoo/odoo/wiki/Contributing#what-does-stable-mean
So, you will need a new idea for that use case :)